cdci_data_analysis/plugins/dummy_instrument/data_server_dispatcher.py

```diff
@@ -60,4 +60,42 @@ def get_dummy_products(self, instrument, config=None, **kwargs):
 
     def process_product_method(self, instrument, prod_list,api=False, **kw):
         query_out = QueryOutput()
-        return query_out
\ No newline at end of file
+        return query_out
+
+    # example with the general user role
+    def check_query_roles(self, roles, par_dic):
+        # if use_max_pointings > 50 or scw_list.split(",") > 50:
+        #     return 'unige-hpc-full' in roles:
+        #
+        # return True
+
+        return 'general' in roles, ['general']
+
+
+class DataServerNumericQuery(ProductQuery):
+
+    def __init__(self, name, parameters_list=[],):
+        super(DataServerNumericQuery, self).__init__(name, parameters_list=parameters_list)
+
+    def test_connection(self):
+        pass
+
+    def test_has_input_products(self):
+        pass
+
+    def get_dummy_products(self, instrument, config=None, **kwargs):
+        return []
+
+    def process_product_method(self, instrument, prod_list, api=False, **kw):
+        query_out = QueryOutput()
+        return query_out
+
+    # example with the general user role
+    def check_query_roles(self, roles, par_dic):
+        if 'p' in par_dic.keys():
+            # not sure this is actually the best way to obtain a certain parameter value
```

Comment:
me neither.
there is this parameter class there, maybe it's good to use that. but it's not critical here.

Reply:
Yes, and that class is actually used later 
[](https://github.com/oda-hub/dispatcher-app/blob/2be8a88c68a621ba1caf462aeb070054c80d2fb0/cdci_data_analysis/analysis/instrument.py#L168)

I will see if it can be used so to make it more uniform and consistent

Reply:
Uses it now, much more consistent and cleaner approach